AmplifyPlugins/DataStore/Tests/AWSDataStorePluginTests/Sync/InitialSync/InitialSyncOrchestratorTests.swift

```diff
@@ -201,6 +202,112 @@ class InitialSyncOrchestratorTests: XCTestCase {
         sink.cancel()
     }
 
+    /// - Given: An InitialSyncOrchestrator with a model dependency graph
+    ///          API is expected to return an unauthorized error for certain models
+    ///          and success for remaining models
+    /// - When:
+    ///    - The orchestrator starts up
+    /// - Then:
+    ///    - It invokes completion callback with success
+    func testInvokesCompletionCallbackWhenUnauthorizedAPIError() async throws {
+        ModelRegistry.reset()
+        PostCommentModelRegistration().registerModels(registry: ModelRegistry.self)
+        let responder = QueryRequestListenerResponder<PaginatedList<AnyModel>> { request, listener in
+            if request.document.contains("SyncPosts") {
+                let urlResponse = HTTPURLResponse(url: URL(string: "https://aws.amazon.com")!,
+                                                 statusCode: 401,
+                                                 httpVersion: nil,
+                                                 headerFields: nil)!
```

Comment:
i think this is good for end-to-end unit testing of the SyncOperation, but this PR adds 3 new checks for categorizing `isUnauthorized`, so there should be 3 APIError states to test. To make it more feasible in testing, the method `isUnauthorizedError` can be made internal, and then the unit test can iterate over an array of APIError types. Ideally, this should be extracted out into its own component and reused across other APIError handling logic in DataStore.